File: domain/src/main/java/com/peratrack/domain/models/Receipt.java
package com.peratrack.domain.models;

import java.util.Date;
import java.util.List;

public final class Receipt {
    public Integer uid;
    public Date date;
    public String storeName;
    public Float totalAmount;
    public List<Product> products;

    public Receipt(
        Integer uid,
        Date date,
        String storeName,
        Float totalAmount,
        List<Product> products
    ) {
        this.uid = uid;
        this.date = date;
        this.storeName = storeName;
        this.totalAmount = totalAmount;
        this.products = products;
    }

    @Override
    public String toString() {
        if(totalAmount != null) {
            return "{" + uid.toString() + " " +
                    date.toString() + " "  +
                    storeName + " "  +
                    totalAmount.toString() + "} ";
        }
        return "{" + uid.toString() + " " +
                date.toString() + " "  +
                storeName + " "  +
                "Total amount is not specified} ";

    }
}

File: data/src/main/java/com/peratrack/data/room/models/Receipt.kt
package com.peratrack.data.room.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "receipt")
data class Receipt(
    @PrimaryKey(autoGenerate = true) val uid: Int = 0,
    val date: Date?,
    val storeName: String?,
    val totalAmount: Float?,
)
